src/features/selectedPost.ts

```diff
@@ -0,0 +1,13 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+const selectedPostSlice = createSlice({
+  name: 'selectedPost',
+  initialState: null as Post | null,
+  reducers: {
+    setSelectedPost: (user, { payload }: PayloadAction<Post | null>) => payload,
```

Comment:
In the `setSelectedPost` reducer, you are returning the payload directly. However, you should be modifying the state, not returning the new state. The correct way would be `state = payload;`. Remember, in Redux Toolkit, we are allowed to 'mutate' the state in reducers, but under the hood, it uses Immer library to handle immutability.